src/components/card-sorting/card-sorting.jsx

```diff
@@ -0,0 +1,56 @@
+import React, {useState, useEffect} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+import {SortingOption} from '../../const';
+import {sortOffers} from '../../store/action';
+import {sortOffersFunc} from '../../utils';
+
+const CardSorting = () => {
+  const [currentFilter, setSortFilter] = useState(`Popular`);
+  const [initialOffers, setInitialOffers] = useState([]);
+  const [optionState, setOptionState] = useState(false);
+  const dispatch = useDispatch();
+
+  const {offers} = useSelector((state) => state.DATA);
+
+  useEffect(() => {
+    if (initialOffers.length === 0) {
+      setInitialOffers(offers);
+    }
+  });
+
+  const handleOpenedFilter = () => {
+    setOptionState(true);
+  };
+
+  const handleClickFilter = ({target}) => {
+    const sortedType = target.dataset.sortedType;
+    const sortedText = target.textContent;
```

Comment:
Давай тут лучше так:
```suggestion
    const sortedText = SortingOption[sortedType];
```
Понадёжней будет. Хотя я в другом месте отписался, что лучше вообще хранить `sortedType` в `currentFilter`.